File: scala/src/main/scala/com/shekhargulati/ninetynine_problems/_01_lists/P22.scala
package com.shekhargulati.ninetynine_problems._01_lists

import scala.annotation.tailrec

/**
  * (*) Create a list containing all integers within a given range
  */
object P22 {

  def range(start: Int, end: Int): List[Int] = start to end toList

  def rangeR(start: Int, end: Int): List[Int] = {
    @tailrec
    def range(list: List[Int], cur: Int): List[Int] = {
      if (cur <= end) {
        range(list :+ cur, cur + 1)
      } else {
        list
      }
    }

    range(List(), start)
  }

}

File: java8/src/main/java/com/shekhargulati/ninetynine_problems/_01_lists/P22.java
package com.shekhargulati.ninetynine_problems._01_lists;

import java.util.List;
import java.util.stream.IntStream;

import static java.util.stream.Collectors.toList;

/**
 * (*) Create a list containing all integers within a given range
 */
public class P22 {

    public static List<Integer> range(int start, int end) {
        return IntStream.rangeClosed(start, end).boxed().collect(toList());
    }
}
